homeassistant/components/firmata/binary_sensor.py

```diff
@@ -0,0 +1,84 @@
+"""Support for Firmata binary sensor input."""
+
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.const import CONF_NAME
+
+from .const import CONF_NEGATE_STATE, CONF_PIN, CONF_PIN_MODE, DOMAIN
+from .pin import FirmataBinaryDigitalInput, FirmataPinUsedException
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Firmata binary sensors."""
+    new_entities = []
+
+    boards = hass.data[DOMAIN]
+    board = boards[config_entry.entry_id]
+    for binary_sensor in board.binary_sensors:
+        try:
+            binary_sensor_entity = FirmataBinarySensor(
+                board, config_entry, **binary_sensor
+            )
+            new_entities.append(binary_sensor_entity)
+        except FirmataPinUsedException:
+            _LOGGER.error(
+                "Could not setup binary sensor on pin %s since pin already in use.",
+                binary_sensor[CONF_PIN],
+            )
+    async_add_entities(new_entities)
+
+
+class FirmataBinarySensor(BinarySensorEntity):
+    """Representation of a binary sensor on a Firmata board."""
+
+    def __init__(self, board, config_entry, **kwargs):
+        """Initialize the binary sensor."""
+        self._name = kwargs[CONF_NAME]
+        self._state = None
+        self._config_entry = config_entry
+        self._conf = kwargs
+
+        pin = self._conf[CONF_PIN]
+        pin_mode = self._conf[CONF_PIN_MODE]
+        negate = self._conf[CONF_NEGATE_STATE]
+
+        self._location = (DOMAIN, self._config_entry.entry_id, "pin", pin)
```

Comment:
`_location`, `_conf`, `_config_entry` and `_state` don't need to be instance attributes.

Reply:
`DOMAIN` doesn't need to be part of the unique_id. The entity registry is aware of both the integration domain and the platform domain.